lib/codecs/src/encoding/format/protobuf.rs

```diff
@@ -64,12 +69,29 @@ pub struct ProtobufSerializerOptions {
 pub struct ProtobufSerializer {
     /// The protobuf message definition to use for serialization.
     message_descriptor: MessageDescriptor,
+    options: Options,
 }
 
 impl ProtobufSerializer {
     /// Creates a new `ProtobufSerializer`.
     pub fn new(message_descriptor: MessageDescriptor) -> Self {
-        Self { message_descriptor }
+        Self {
+            message_descriptor,
+            options: Options::default(),
+        }
+    }
+
+    /// Creates a new serializer instance using the descriptor bytes directly.
+    pub fn new_from_bytes(
+        desc_bytes: &[u8],
+        message_type: &str,
+        options: &Options,
```

Comment:
Nit: I'd expect this to take an owned `Options` since it's a new function.

```suggestion
        options: Options,
```